src/prep_cif.c

```diff
@@ -234,7 +234,7 @@ ffi_status ffi_prep_cif_var(ffi_cif *cif,
   return ffi_prep_cif_core(cif, abi, 1, nfixedargs, ntotalargs, rtype, atypes);
 }
 
-#if FFI_CLOSURES
+#if FFI_CLOSURES && FFI_LEGACY_CLOSURE_API
```

Comment:
What is FFI_LEGACY_CLOSURE_API?

Reply:
In our build, it's defined in our ffitarget.hs as:

darwin/include/ffitarget_arm64.h
54:#define FFI_LEGACY_CLOSURE_API 0

darwin/include/ffitarget_armv7.h
68:#define FFI_LEGACY_CLOSURE_API 0

darwin/include/ffitarget_x86.h
125:#define FFI_LEGACY_CLOSURE_API 1

We just didn't want to make ffi_prep_closure() available on arm since it won't work there... better to catch issues at build time than at runtime.  We should omit these bits from the final PR.